src/sagemaker/jumpstart/curated_hub/content_copy.py

```diff
@@ -0,0 +1,289 @@
+        markdown_s3_reference_dst = self._dst_s3_filesystem.get_markdown_s3_reference(model_specs)
+        copy_configs.append(
+            CopyContentConfig(
+                src=markdown_s3_reference, dst=markdown_s3_reference_dst, display_name="markdown"
+            )
+        )
+
+        return copy_configs
+
+    def _parallel_execute_copy_configs(self, copy_configs: List[CopyContentConfig]):
+        tasks = []
+        with futures.ThreadPoolExecutor(
+            max_workers=self._thread_pool_size, thread_name_prefix="import-models-to-curated-hub"
+        ) as deploy_executor:
+            for copy_config in copy_configs:
+                tasks.append(
+                    deploy_executor.submit(
+                        self._copy_s3_reference,
+                        copy_config.display_name,
+                        copy_config.src,
+                        copy_config.dst,
+                    )
+                )
+
+        results = futures.wait(tasks)
+        failed_copies: List[BaseException] = []
+        for result in results.done:
+            exception = result.exception()
+            if exception:
+                failed_copies.append(exception)
+            if failed_copies:
+                raise RuntimeError(
+                    f"Failures when importing models to curated hub in parallel: {failed_copies}"
+                )
+
+    def _copy_s3_reference(
+        self, resource_name: str, src: S3ObjectReference, dst: S3ObjectReference
+    ):
+        if not self._is_s3_object_different(src, dst):
+            print(
+                f"Detected that {resource_name} is the same in destination bucket. Skipping copy."
+            )
+            return
+
+        print(f"Copying {resource_name} from {src.bucket}/{src.key} to {dst.bucket}/{dst.key}...")
+        try:
+            self._s3_client.copy(
+                src.format_for_s3_copy(),
+                dst.bucket,
+                dst.key,
+                ExtraArgs=EXTRA_S3_COPY_ARGS,
+            )
+        except Exception as ex:
+            print(
+                "ERROR: encountered an exception when calling copy from"
+                + f" {src.bucket}/{src.key} to {dst.bucket}/{dst.key}: {str(ex)}"
+            )
+            raise ex
+
+        print(
+            f"Copying {resource_name} from {src.bucket}/{src.key} to {dst.bucket}/{dst.key} complete!"
+        )
+
+    def _is_s3_object_different(self, src: S3ObjectReference, dst: S3ObjectReference) -> bool:
+        src_etag = self._get_s3_object_etag(src)
+        dst_etag = self._get_s3_object_etag(dst)
+
+        return src_etag != dst_etag
+
+    def _get_s3_object_etag(self, object: S3ObjectReference) -> str:
+        try:
+            response = self._s3_client.head_object(Bucket=object.bucket, Key=object.key)
+            return response.pop("ETag")
+        except Exception:
+            return ""
```

Comment:
For this case, can we return `None`? That seems cleaner than empty string

Reply:
+1